components/category_menu_items/category_menu_items.tsx

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {injectIntl, IntlShape} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {CategoryTypes} from 'mattermost-redux/constants/channel_categories';
+import {getCategoryInTeamWithChannel} from 'mattermost-redux/selectors/entities/channel_categories';
+import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
+import {Channel} from 'mattermost-redux/types/channels';
+import {DispatchFunc} from 'mattermost-redux/types/actions';
+import {ChannelCategory} from 'mattermost-redux/types/channel_categories';
+
+import {trackEvent} from 'actions/telemetry_actions';
+import {addChannelsInSidebar} from 'actions/views/channel_sidebar';
+import {openModal} from 'actions/views/modals';
+import EditCategoryModal from 'components/edit_category_modal';
+import Menu from 'components/widgets/menu/menu';
+import {getCategoriesForCurrentTeam, getDisplayedChannels} from 'selectors/views/channel_sidebar';
+import {GlobalState} from 'types/store';
+import Constants, {ModalIdentifiers} from 'utils/constants';
+
+type Props = {
+    channel: Channel;
+    intl: IntlShape;
+    openUp: boolean;
+};
+
+const CategoryMenuItems = (props: Props): JSX.Element | null => {
+    const {channel, intl, openUp} = props;
+    const dispatch = useDispatch<DispatchFunc>();
+    const displayedChannels = useSelector((state: GlobalState) => getDisplayedChannels(state));
+    const multiSelectedChannelIds = useSelector((state: GlobalState) => state.views.channelSidebar.multiSelectedChannelIds);
+    const currentTeam = useSelector((state: GlobalState) => getCurrentTeam(state));
+
+    let categories: ChannelCategory[] | undefined;
+    let currentCategory: ChannelCategory | undefined;
+
+    if (currentTeam) {
+        categories = useSelector((state: GlobalState) => getCategoriesForCurrentTeam(state));
+        currentCategory = useSelector((state: GlobalState) => getCategoryInTeamWithChannel(state, currentTeam.id, channel.id));
+    }
+
+    const moveToCategory = useCallback((categoryId: string) => {
+        dispatch(addChannelsInSidebar(categoryId, channel.id));
+        trackEvent('ui', 'ui_sidebar_channel_menu_moveToExistingCategory');
+    }, []);
```

Comment:
This needs to list `channel` as a dependency of the callback

Reply:
Actually, since this isn't passed to a child component as a prop, I don't think this even needs a `useCallback`